sdk/core/azure-core/inc/azure/core/core.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+/**
+ * @file core.hpp
+ * @brief Adds all headers from Azure Core.
+ *
+ */
+
+// azure/core
+#include "azure/core/azure.hpp"
+#include "azure/core/context.hpp"
+#include "azure/core/credentials.hpp"
+#include "azure/core/datetime.hpp"
+#include "azure/core/nullable.hpp"
+#include "azure/core/response.hpp"
+#include "azure/core/uuid.hpp"
+#include "azure/core/version.hpp"
+
+// azure/core/http
+#include "azure/core/http/body_stream.hpp"
+#include "azure/core/http/http.hpp"
+#include "azure/core/http/pipeline.hpp"
+#include "azure/core/http/policy.hpp"
+#include "azure/core/http/transport.hpp"
+
+// azure/core/http/curl
+#include "azure/core/http/curl/curl.hpp"
+
+// azure/core/http/winhttp
+#include "azure/core/http/winhttp/win_http_client.hpp"
```

Comment:
Transport adapters should be included in CMake but not within the core header

Reply:
why?
its public header...

Reply:
It shouldn't be included by default, right? The user picks the transport adapter and based on that includes the specific header they need. We don't want the main core.hpp to include optional things that are configuration based, especially transitively (like curl.hpp).

Reply:
curl and winhttp headers contains 
#ifdef BUILD_CURL_HTTP_TRANSPORT_ADAPTER
and
#ifdef BUILD_TRANSPORT_WINHTTP_ADAPTER

The way we have CMake set up, we don't add headers explicitly, we add the entire azure/inc folder. So, CMake can only control adding the source files (cpp), but the header is always there.
We make the header to be empty with the options.

So, the simplified header would always include both transport adapters, and only provide APIs when they are BUILT

Reply:
> So, CMake can only control adding the source files (cpp), but the header is always there.

I understand. I was saying that the user can explicitly add the specific transport header they want to include.

> We make the header to be empty with the options.
> So, the simplified header would always include both transport adapters, and only provide APIs when they are BUILT

OK, that makes sense. Thanks.

Reply:
> I understand. I was saying that the user can explicitly add the specific transport header they want to include.

Yes, however, we want to use this simplified header for the API Review tool, and we want to have the transport adapters APIs to be there as well.

> We could add these ifdefs in the core.hpp too, right? So we conditionally include only the relevant headers. Any reason not to do that?

Yes, but we would be duplicating the validation (doing the same thing). To me it feels like it's not required since we are already covering it from the header itself. I mean, it would be no difference on adding the condition again.

Reply:
> Yes, however, we want to use this simplified header for the API Review tool, and we want to have the transport adapters APIs to be there as well.

Let's not include point-in-time/one-off APIview generation/process related reasons as part of the simplified header design. If necessary, we could modify that behavior locally, and not have that file checked-in.

> Yes, but we would be duplicating the validation (doing the same thing). To me it feels like it's not required since we are already covering it from the header itself. I mean, it would be no difference on adding the condition again.

Sure, that's fine. I agree the difference would be subtle. We can revisit that later if needed.